File: tests/TranscribeTest.java
import java.io.File;
import java.io.IOException;
import java.time.Duration;
import java.time.Instant;
import okhttp3.*;

public class TranscribeTest {
    private static final String BASE_URL = "http://127.0.0.1:8080";
    private static final String ENDPOINT = "/transcribe";

    public static void main(String[] args) {
        testTranscribeApi();
    }

    public static void testTranscribeApi() {
        // 测试音频文件
        File audioFile = new File("tests/test_files/test_audio.mp3");
        if (!audioFile.exists()) {
            System.out.println("测试音频文件不存在: " + audioFile.getPath());
            return;
        }

        try {
            OkHttpClient client = new OkHttpClient();

            // 构建multipart请求
            RequestBody requestBody = new MultipartBody.Builder()
                .setType(MultipartBody.FORM)
                .addFormDataPart("file", "test_audio.mp3",
                    RequestBody.create(MediaType.parse("audio/mpeg"), audioFile))
                .addFormDataPart("language_code", "zh-CN")
                .addFormDataPart("model", "latest_long")
                .build();

            Request request = new Request.Builder()
                .url(BASE_URL + ENDPOINT)
                .post(requestBody)
                .build();

            // 记录开始时间
            Instant start = Instant.now();

            // 发送请求
            Response response = client.newCall(request).execute();
            String responseBody = response.body().string();

            // 计算耗时
            Duration duration = Duration.between(start, Instant.now());

            // 打印测试结果
            System.out.println("\n=== 转写接口测试结果 ===");
            System.out.println("请求耗时: " + duration.toMillis() / 1000.0 + "秒");
            System.out.println("状态码: " + response.code());
            System.out.println("响应内容: " + responseBody);

            if (response.code() == 200) {
                System.out.println("\n✓ 测试通过！");
            } else {
                System.out.println("\n✗ 测试失败！");
            }

        } catch (IOException e) {
            System.out.println("\n✗ 测试失败: " + e.getMessage());
        }
    }
}

File: tests/transcribe-test.js
const fs = require('fs');
const axios = require('axios');
const FormData = require('form-data');

const BASE_URL = 'http://127.0.0.1:8080';
const ENDPOINT = '/transcribe';

async function testTranscribeApi() {
    // 测试音频文件路径
    const audioFilePath = 'tests/test_files/test_audio.mp3';

    try {
        // 检查文件是否存在
        if (!fs.existsSync(audioFilePath)) {
            throw new Error(`测试音频文件不存在: ${audioFilePath}`);
        }

        // 准备表单数据
        const formData = new FormData();
        formData.append('file', fs.createReadStream(audioFilePath));
        formData.append('language_code', 'zh-CN');
        formData.append('model', 'latest_long');

        // 记录开始时间
        const startTime = Date.now();

        // 发送请求
        const response = await axios.post(`${BASE_URL}${ENDPOINT}`, formData, {
            headers: {
                ...formData.getHeaders()
            }
        });

        // 计算耗时
        const duration = (Date.now() - startTime) / 1000;

        // 打印测试结果
        console.log('\n=== 转写接口测试结果 ===');
        console.log(`请求耗时: ${duration.toFixed(2)}秒`);
        console.log(`状态码: ${response.status}`);
        console.log(`响应内容:`, response.data);

        if (response.status === 200 && response.data.transcription) {
            console.log('\n✓ 测试通过！');
        } else {
            console.log('\n✗ 测试失败：响应格式错误');
        }

    } catch (error) {
        console.log('\n✗ 测试失败:', error.message);
        if (error.response) {
            console.log('错误响应:', error.response.data);
        }
    }
}

// 运行测试
testTranscribeApi();